lib/ansible/module_utils/vmware.py

```diff
@@ -431,14 +434,57 @@ def list_snapshots(vm):
     return result
 
 
+def get_vnc_extraconfig(vm):
+    result = {}
+    extraconfig = vm.config.extraConfig
+    for opts in extraconfig:
+        for optkeyname in ['enabled', 'ip', 'port', 'password']:
+            if opts.key.lower() == "remotedisplay.vnc." + optkeyname:
+                result[optkeyname] = opts.value
+    return result
+
+
+def set_vnc_extraconfig(content, vm, enabled, ip, port, password):
+    result = dict(
+        changed=False,
+        failed=False,
+    )
+    spec = vim.vm.ConfigSpec()
+    spec.extraConfig = []
+    options_values = {
+        "remotedisplay.vnc.enabled": "",
+        "remotedisplay.vnc.ip": "",
+        "remotedisplay.vnc.port": "",
+        "remotedisplay.vnc.password": ""}
+    if enabled:
+        options_values["remotedisplay.vnc.enabled"] = "true"
+        options_values["remotedisplay.vnc.password"] = str(password).strip()
+        options_values["remotedisplay.vnc.ip"] = str(ip).strip()
+        options_values["remotedisplay.vnc.port"] = str(port).strip()
+    for k, v in options_values.items():
+        opt = vim.option.OptionValue()
+        opt.key = k
+        opt.value = v
+        spec.extraConfig.append(opt)
+    task = vm.ReconfigVM_Task(spec)
+    wait_for_task(task)
+    if task.info.state == 'error':
+        result['failed'] = True
+        result['msg'] = task.info.error.msg
+    else:
+        result['changed'] = True
+        result['instance'] = gather_vm_facts(content, vm)
+    return result
+
+
 def vmware_argument_spec():
 
     return dict(
         hostname=dict(type='str', required=True),
         username=dict(type='str', aliases=['user', 'admin'], required=True),
         password=dict(type='str', aliases=['pass', 'pwd'], required=True, no_log=True),
         port=dict(type='int', default=443),
-        validate_certs=dict(type='bool', required=False, default=True),
+        validate_certs=dict(type='bool', required=False, default=True, choices=[True, False]),
```

Comment:
Validate_certs can accept 'yes' or 'no'

Reply:
I suspected that, and tested them after the change. yes/no works on all playbooks. I guess Ansible maps yes/no to true/false somewhere for bool type.
Tests were failing and I had to add that "choices" option there. That was a fix for failing tests :)